### ISSUE TITLE
[Terminal Chat] The LLM receives confusing information about shell

### ISSUE THREAD

**Issue by vbrozik** [microsoft/terminal#18142]
### Windows Terminal version

1.23.3061.0

### Windows build number

10.0.22631.0

### Other Software

- Ubuntu 22.04.5 LTS - installed as the default Ubuntu distribution in WLS2, then gradually updated
- GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)

### Steps to reproduce

1. Use the default Ubuntu distribution in WSL.
2. Open a terminal for it.
3. Start a Terminal Chat.
4. After few interactions it could mention that it has information that I am using a shell named `ubuntu.exe`:

![Image](https://github.com/user-attachments/assets/efd246b4-67bf-4f0e-943b-979006b409e9)

### Expected Behavior

The chatbot should not behave is if it received wrong information about the shell being used. It should either have information that I am using Ubuntu distribution of Linux (and maybe version) or that I am using Bash shell in Ubuntu Linux.

### Actual Behavior

The Terminal Chat probably receives confusing information about the shell in the terminal window in its preconfigured initial prompt.

When I am using the default Ubuntu WSL distribution with the default `bash` shell the GitHub Copilot responds as if it received information that I am using shell named `ubuntu.exe`:

![Image](https://github.com/user-attachments/assets/efd246b4-67bf-4f0e-943b-979006b409e9)

**Comment by zcobol** [comment]
@vbrozik LLM hallucination 😀

**Comment by vbrozik** [comment]
@zcobol we can see the hallucination in the second case: `ubuntu2404.exe` vs "Ubuntu 20.04"

...but I think we can be almost sure that the LLM receives information about the command used to start the WSL2 session like `ubuntu.exe` or `ubuntu2404.exe` and it is being wrongly presented to it as a shell running inside the terminal.

**Comment by DHowett** [comment]
This is one of the annoying limitations we've run into - Terminal actually cannot know what is running "behind" the single executable that it started (in this case, `ubuntu.exe` or `ubuntu2404.exe`). There's no interface by which this data is communicated, so it makes a "best guess" and inserts the name of the root process of the tab.

It's not ideal.

**Comment by vbrozik** [comment]
Yes, I expected that Terminal lacking the information was the cause. I think that presenting `ubuntu.exe` to LLMs as a shell could cause bad experience of the users. I really do not know what is optimal to present to LLMs but I would certainly consider one or more of following solutions:

- Present the command like `ubuntu.exe` as a command used to start the environment inside the Terminal, not as a shell.
- Alternatively do not present the command at all to the LLM.
- Create a configuration of initial prompt for the LLMs a part of the profile configuration. One of the parameters would be the default shell used with the profile. It will be the place to specify `bash`, `POSIX shell`, `PowerShell` etc.
- When a profile is created the initial value of the default shell could be automatically inferred. Example: `ubuntu.exe` -> `bash` or `bash in Ubuntu`; `powershell.exe` -> `PowerShell`
- Later there could be some kind of integration of the Terminal with shells. It would be really good to cooperate with the VS Code terminal team. For example the shell could communicate about itself to the Terminal using control sequences. This way the Terminal Chat could work correctly if you start a new shell even on a different machine for example through SSH.

**Comment by PankajBhojwani** [comment]
Those are some solid suggestions, thank you! We actually do have a work item for allowing the user to modify/add to the system prompt that we send to the LLM, once that is implemented we can remove the 'default' addition that we do right now entirely and just use the user-provided one, which should fix this :)

**Comment by DHowett** [comment]
We recently announced a new "Intelligent Terminal", which is the evolution of Terminal Chat. It's currently positioned as a totally separate product to make it clear that it's not ready to roll out to all of our many users. :)

Check it out at http://aka.ms/intelligent-terminal
